### PR TITLE
Construct pages with routing

### PR DESCRIPTION
**Pull request by Darren52BT**
This addresses #139.

# What I've Done
I have basically created a shell for the website.  I used React Router to create routes for each page. Upon loading, you should see a temporary list of links that I put down to make it easier to navigate around these routes. You should also be able to click on links on various pages to go to their respective pages. I've only done so for links that we have page components for, so the other links do nothing.

# Structure 
Currently, there are three sets of routes that are stacked on top of each other. The first defines routes for which header to display, the second for which main page component to display ~~, and the third for which footer to display.~~  (Edit: I've changed it to always display the footer with links, and to show the messages appbar fixed at the bottom if we're at a logged out page). I renamed and moved the various header and footer components to separate folders to make it more organized. I also created another header component for the logged out pages aside from the logged out home page.

All route paths are being defined within an object called paths that is being used globally to reference to a certain path. If we want to change the path for a certain page globally, all we need to do is edit this object.

I've edited the various components representing links to function as the Link component from React Router.

# Notes
-as this is a shell, this is just showing which components to display for the current route. We will need to rework it further later to pass data between pages.
-I think that there is supposed to be a separate page for the forgot password page. I have defined a page for it but it does not completely fill up the screen so it looks pretty awkward. Not sure if I should take care of it here or if that should be done separately.

# What You Should see
## Default Page (containing temp link list)
image

## Ex: Clicking on Logged Out Home Page Link:
image

image

## Ex: Clicking on Swiping Link in Logged In Header:
image
image

### CODE REVIEW

File: meshapp/src/Routing/FooterRoutes.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  paths.sign_up,
+];
+
+//contains routes for the footer
+export default function FooterRoutes() {
+  return (
+    <Routes>
+      {LOGGED_IN_PATHS.map((path) => (
+        <Route path={path} element={<LoggedInFooter />} />
+      ))}
```
Comment (misslame) [inline meshapp/src/Routing/FooterRoutes.tsx:25]: We have different footers based on login?

File: meshapp/src/Routing/HeaderRoutes.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const LOGGED_OUT_PATHS = [
+  //   paths.logged_out_home,
+  paths.login_page,
+  paths.forgot_password,
+  paths.sign_up,
+];
+
+export default function HeaderRoutes() {
+  return (
+    <Routes>
```
Comment (misslame) [inline meshapp/src/Routing/HeaderRoutes.tsx:21]: Something seems fishy but idk how to fix, why are header and footer files essentially the same?

File: meshapp/src/home/login.tsx [hunk 3]
```diff
@@ -73,20 +78,21 @@ const Login = () => {
             bgcolor: "cardBackground.main",
             color: "text.main",
             borderRadius: 5,
-          }}>
+          }}
+        >
```
Comment (misslame) [inline meshapp/src/home/login.tsx:82]: Nit picky but why new line

**Review by misslame — APPROVED** [review]
Approving but I have comments

File: meshapp/src/Routing/HeaderRoutes.tsx (hunk 2 above)
Comment (Darren52BT) [reply meshapp/src/Routing/HeaderRoutes.tsx:21]: Well they are pretty much doing the same thing, which is showing a specific version of the header/footer for logged in/logged out pages, but the header for the logged out home page is different from the other logged out pages, so there's a difference in routes.

File: meshapp/src/Routing/FooterRoutes.tsx (hunk 1 above)
Comment (Darren52BT) [reply meshapp/src/Routing/FooterRoutes.tsx:25]: The footer with the message icon is different from the one with all the links right? That's what I'm seeing on the figma

File: meshapp/src/Routing/MainPageRoutes.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import LoggedOutPage from "../home/logged-out/home";
+import SignUp from "../components/SignUp/SignUp";
+import Login from "../home/login";
+import ForgotPassword from "../components/password-forms/forgot-password-form";
+import LoggedInHome from "../home/logged-in/LoggedInHome";
+import ProfilePage from "../profile/profile-page";
+import {
+  exampleProfile,
+  exampleProfile2,
```
Comment (JRSiwiecki) [inline meshapp/src/Routing/MainPageRoutes.tsx:9]: Is exampleProfile2 being used anywhere else here?

**Review by JRSiwiecki — COMMENTED** [review]
Everything seems mostly fine to me, I just have a question: when I was viewing the site on your branch, it seemed that for some pages like the forgot_password page or signup page, the home link doesn't lead anywhere, is that correct? I can understand if that's just confusion over whether to send people to the logged in home or logged out home, but I think as a nice default it would just send people to logged out home (or even back to the main routing page).

Other than those questions, I really don't have much to say other than good work! Please re-ping me on discord so I can send approval! 🥇

File: meshapp/src/Routing/FooterRoutes.tsx (hunk 1 above)
Comment (misslame) [reply meshapp/src/Routing/FooterRoutes.tsx:25]: The message icon shouldn't be a part of the footer it should be by itself

File: meshapp/src/Routing/HeaderRoutes.tsx (hunk 2 above)
Comment (misslame) [reply meshapp/src/Routing/HeaderRoutes.tsx:21]: Hmmmm

Comment (misslame) [reply meshapp/src/Routing/HeaderRoutes.tsx:21]: Ill come back to this

File: meshapp/src/Routing/FooterRoutes.tsx (hunk 1 above)
Comment (Darren52BT) [reply meshapp/src/Routing/FooterRoutes.tsx:25]: It's shown consistently along all the logged in pages at the bottom though so shouldnt it be a footer? should the footer with the links be included in the logged in pages as well?

Comment (misslame) [reply meshapp/src/Routing/FooterRoutes.tsx:25]: If only one thing is different, the whole footer shouldn't be rendered as a different footer

Comment (misslame) [reply meshapp/src/Routing/FooterRoutes.tsx:25]: Its showed down there. It should be like the facebook message bar

Comment (Darren52BT) [reply meshapp/src/Routing/FooterRoutes.tsx:25]: So is it meant to be something that is fixed at the bottom and appears over other items similarly to the 2FA banner? In what other ways should I be displaying this for the logged in pages?

**Pull request merged**